File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        int pre = 1;
        res[0] = 1;
        for(int i=1;i<nums.length;i++)
        {
            res[i] = nums[i-1]*res[i-1];
        }

        int post = nums[nums.length-1];
        for(int i=nums.length-2;i>=0;i--)
        {
            res[i] = res[i]*post;
            post = post*nums[i];
        }
        return res;
    }
}  

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int res;
        int start =0, end = nums.size()-1;
        while(start<=end)
        {
            int mid = start + (end-start) / 2 ;

            if(nums[start]<nums[end])
            {
                res = min(res,nums[start]);
                return res;
            }

            res = min(res,nums[mid]);

            if(nums[start]>nums[mid])
            {
                end = mid-1;
            }
            else 
            {
                start = mid+1;
            }


        }

    return res;
    }
};
